plugins/scaffolder/src/next/OngoingTask/TaskSteps/StepTime.tsx

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React, { useCallback, useMemo, useState } from 'react';
+import useInterval from 'react-use/lib/useInterval';
+import { DateTime, Interval } from 'luxon';
+import humanizeDuration from 'humanize-duration';
+import { Typography } from '@material-ui/core';
+
+export const StepTime = (props: {
+  step: {
+    id: string;
+    name: string;
+    startedAt?: string;
+    endedAt?: string;
+  };
+}) => {
+  const [time, setTime] = useState('');
+  const { step } = props;
+
+  const calculate = useCallback(() => {
+    if (!step.startedAt) {
+      setTime('');
+      return;
+    }
+
+    const end = step.endedAt
+      ? DateTime.fromISO(step.endedAt)
+      : DateTime.local();
+
+    const startedAt = DateTime.fromISO(step.startedAt);
+    const formatted = Interval.fromDateTimes(startedAt, end)
+      .toDuration()
+      .valueOf();
+
+    setTime(humanizeDuration(formatted, { round: true }));
+  }, [step.endedAt, step.startedAt]);
+
+  useMemo(() => calculate(), [calculate]);
```

Comment:
Matter of taste, but how do you like this

```ts
const [counter, setCounter] = useState(0);

const time = useMemo(
  () => { ...compute... },
  [step.startedAt, step.endedAt, (!step.endedAt && counter)]
);

useInterval(() => setCounter(c => c + 1), 1000);
```

it just feels odd to (1) depend on the instance value of a function, (2) use useMemo for what's effectively an effect, and (3) have a callback to effectively do setState, as a way of achieving memoization instead of using useMemo

Reply:
i'm feeling it might be better off to work out if we can drop the `useMemo` entirely for readability sake. Like `useMount(() => calculate())` or something